lib/itamae/resource/file.rb

```diff
@@ -84,23 +84,27 @@ def action_delete(options)
       end
 
       def action_edit(options)
+        change_target = attributes.modified ? @temppath : attributes.path
+
         if attributes.mode
-          run_specinfra(:change_file_mode, @temppath, attributes.mode)
+          run_specinfra(:change_file_mode, change_target, attributes.mode)
         else
           mode = run_specinfra(:get_file_mode, attributes.path).stdout.chomp
-          run_specinfra(:change_file_mode, @temppath, mode)
+          run_specinfra(:change_file_mode, change_target, mode)
         end
 
         if attributes.owner || attributes.group
-          run_specinfra(:change_file_owner, @temppath, attributes.owner, attributes.group)
+          run_specinfra(:change_file_owner, change_target, attributes.owner, attributes.group)
         else
           owner = run_specinfra(:get_file_owner_user, attributes.path).stdout.chomp
           group = run_specinfra(:get_file_owner_group, attributes.path).stdout.chomp
-          run_specinfra(:change_file_owner, @temppath, owner)
-          run_specinfra(:change_file_group, @temppath, group)
+          run_specinfra(:change_file_owner, change_target, owner)
```

Comment:
I understand following:

* Case: `attributes.modified` is true.
    * `change_target` is `@temppath`.
    * need `:change_file_owner`.
* Case: `attributes.modified` is false.
    * `change_target` is `attributes.path`.
    * DON'T need `:change_file_owner`.

So I must fix to following:

```diff
diff --git a/lib/itamae/resource/file.rb b/lib/itamae/resource/file.rb
index d938523..1d26af7 100644
--- a/lib/itamae/resource/file.rb
+++ b/lib/itamae/resource/file.rb
@@ -95,11 +95,11 @@ def action_edit(options)
 
         if attributes.owner || attributes.group
           run_specinfra(:change_file_owner, change_target, attributes.owner, attributes.group)
-        else
+        elsif attributes.modified
           owner = run_specinfra(:get_file_owner_user, attributes.path).stdout.chomp
           group = run_specinfra(:get_file_owner_group, attributes.path).stdout.chomp
-          run_specinfra(:change_file_owner, change_target, owner)
-          run_specinfra(:change_file_group, change_target, group)
+          run_specinfra(:change_file_owner, @temppath, owner)
+          run_specinfra(:change_file_group, @temppath, group)
         end
 
         if attributes.modified
```

Is it OK?

(`elsif attributes.modified` removes extra `run_specinfra` calling)